components/stack/stack.vue

```diff
@@ -0,0 +1,204 @@
+<template>
+  <component
+    :is="stackElement"
+    :class="[
+      'd-stack',
+      stackResponsive(),
+      defaultDir(),
+      stackGap,
+    ]"
+  >
+    <!-- @slot Slot for main content -->
+    <slot />
+  </component>
+</template>
+
+<script>
+import { DT_STACK_DIR, DT_STACK_GAP, DT_STACK_RESPONSIVE_BREAKPOINTS } from './stack_constants';
+
+export default {
+  name: 'DtStack',
+
+  props: {
+    /**
+     * Set this prop to the direction to stack the items.
+     * You can override the default direction with 'default' key.
+     * All the undefined breakpoints will have 'default' value
+     */
+    dir: {
+      type: [String, Object],
+      default: 'column',
+    },
+
+    /**
+     * Set this prop to render stack as a specific HTML element.
+     */
+    as: {
+      type: String,
+      default: 'div',
+    },
+
+    /**
+     * Set this prop to have the space between each stack item
+     */
+    gap: {
+      type: String,
+      default: '0',
+    },
+  },
+
+  data () {
+    return {
+      DT_STACK_DIR,
+      DT_STACK_GAP,
+      DT_STACK_RESPONSIVE_BREAKPOINTS,
+    };
+  },
+
+  computed: {
+    stackDir () {
+      return this.dir;
+    },
+
+    stackGap () {
+      return DT_STACK_GAP[this.gap] ? `d-stack--gap-${this.gap}` : null;
+    },
+
+    stackElement () {
+      return this.as ? this.as : 'div';
+    },
```

Comment:
It was just to quick fix reactivity on the storybook side.
If you remove the `div` value on the `as` prop, it will not render anything.

If you use this fallback, it will not detect nothing and then it will render a `div`.

Im also agree to just use `as` directly on the component.

Reply:
You sure this is the case? just tested locally with `:is="as"` and got this

![2023-02-15 16 20 06](https://user-images.githubusercontent.com/64808812/219226027-806ad347-4d17-4330-8bee-5f56cbb480ec.gif)

Reply:
https://user-images.githubusercontent.com/89984179/219400002-64ef507c-da7a-4961-81ee-ea3381d94e88.mov

Reply:
Ah I see it's only when it's completely blank. Does it fix it if we add

```
stackElement () {
  return this.as ? this.as : 'div';
},
```

to `stack_default.story.vue` rather than the component itself?

Reply:
Ye sure, I didn't thing on the possibility of manipulate a little bit the default.story.
Im seeing you guys are doing this kind of things to not contaminate the component :)